altair/theme.py

```diff
@@ -0,0 +1,290 @@
+    "TitleParamsKwds",
+    "TooltipContentKwds",
+    "TopLevelSelectionParameterKwds",
+    "VariableParameterKwds",
+    "ViewBackgroundKwds",
+    "ViewConfigKwds",
+    "active",
+    "enable",
+    "get",
+    "names",
+    "options",
+    "register",
+    "themes",
+    "unregister",
+]
+
+
+def register(
+    name: LiteralString, *, enable: bool
+) -> Callable[[Plugin[ThemeConfig]], Plugin[ThemeConfig]]:
+    """
+    Decorator for registering a theme function.
+
+    Parameters
+    ----------
+    name
+        Unique name assigned in ``alt.theme.themes``.
+    enable
+        Auto-enable the wrapped theme.
+
+    Examples
+    --------
+    Register and enable a theme::
+
+        import altair as alt
+        from altair import theme
+
+
+        @theme.register("param_font_size", enable=True)
+        def custom_theme() -> theme.ThemeConfig:
+            sizes = 12, 14, 16, 18, 20
+            return {
+                "autosize": {"contains": "content", "resize": True},
+                "background": "#F3F2F1",
+                "config": {
+                    "axisX": {"labelFontSize": sizes[1], "titleFontSize": sizes[1]},
+                    "axisY": {"labelFontSize": sizes[1], "titleFontSize": sizes[1]},
+                    "font": "'Lato', 'Segoe UI', Tahoma, Verdana, sans-serif",
+                    "headerColumn": {"labelFontSize": sizes[1]},
+                    "headerFacet": {"labelFontSize": sizes[1]},
+                    "headerRow": {"labelFontSize": sizes[1]},
+                    "legend": {"labelFontSize": sizes[0], "titleFontSize": sizes[1]},
+                    "text": {"fontSize": sizes[0]},
+                    "title": {"fontSize": sizes[-1]},
+                },
+                "height": {"step": 28},
+                "width": 350,
+            }
+
+    Until another theme has been enabled, all charts will use defaults set in ``custom_theme``::
+
+        from vega_datasets import data
+
+        source = data.stocks()
+        lines = (
+            alt.Chart(source, title=alt.Title("Stocks"))
+            .mark_line()
+            .encode(x="date:T", y="price:Q", color="symbol:N")
+        )
+        lines.interactive(bind_y=False)
+
+    """
+
+    # HACK: See for `LiteralString` requirement in `name`
+    # https://github.com/vega/altair/pull/3526#discussion_r1743350127
+    def decorate(func: Plugin[ThemeConfig], /) -> Plugin[ThemeConfig]:
+        themes.register(name, func)
+        if enable:
+            themes.enable(name)
+
+        @_wraps(func)
+        def wrapper(*args: P.args, **kwargs: P.kwargs) -> ThemeConfig:
+            return func(*args, **kwargs)
+
+        return wrapper
+
+    return decorate
+
+
+def unregister(name: LiteralString) -> Plugin[ThemeConfig] | None:
+    """
+    Remove and return a previously registered theme.
+
+    Parameters
+    ----------
+    name
+        Unique name assigned in ``alt.theme.themes``.
+    """
+    return themes.register(name, None)
```

Comment:
# Note
- More explicit than being part of `themes.register`
- This behavior is **not part of `@alt.theme.register`**
- It wouldn't make sense decorating a function with it